index.bs

```diff
@@ -485,6 +521,155 @@ spec:csp3; type:grammar; text:base64-value
+  1. If |headers| <a for="header list">contains</a> a <a>header</a>
+     whose <a>header name</a> is "Integrity-Policy-Report-Only",
+     let |container|'s <a>report only integrity policy</a> be the result of running
+     <a>processing an integrity policy</a> with the corresponding <a>header value</a>.
+  
+  ### Should request be blocked by Integrity Policy ### {#should-request-be-blocked-by-integrity-policy-section}
+  To determine <dfn>should request be blocked by integrity policy</dfn>, given a <a>request</a> |request|,
+  do the following:
+
+  1. Let |policy container| be |request|'s <a>policy container</a>.
+  1. if |request|'s <a>integrity metadata</a> is not the empty string
+     and |request|'s <a for=request>mode</a> is either "cors" or "same-origin",
+     return "Allowed".
+  1. Let |policy| be |policy container|'s <a>integrity policy</a>.
+  1. Let |report policy| be |policy container|'s <a>report only integrity policy</a>.
+  1. If both |policy| and |report policy| are empty <a>integrity policy struct</a>s, return "Allowed".
+  1. Let |global| be |request|'s <a for=request>client</a>'s <a>global object</a>.
+  1. if |global| is not a <a>Window</a> nor a <a>WorkerGlobalScope</a>, return "Allowed".
+  1. Let |block| be a boolean, initially false.
+  1. Let |report block| be a boolean, initially false.
+  1. if |policy|'s <a>sources</a> <a for=list>contains</a> "inline"
+     and |policy|'s <a>blocked destinations</a> <a for=list>contains</a>
+     |request|'s <a for=request>destination</a>,
+     set |block| to true.
+  1. if |policy|'s <a>sources</a> <a for=list>contains</a> "inline"
+     and |report policy|'s <a>blocked destinations</a> <a for=list>contains</a>
+     |request|'s <a for=request>destination</a>,
+     set |report block| to true.
+  1. If |block| or |report block|, <a>report violation</a> with |request|, |block|, |report block|,
+     |policy| and |report policy|.
+  1. If |block|, return "Blocked". Otherwise, return "Allowed".
+ 
+  ### Report violations ### {#report-violations}
+
+  <pre class="idl">
+    [Exposed=Window]
+    interface IntegrityPolicyViolationReportBody : ReportBody {
+      [Default] object toJSON();
+      readonly attribute USVString documentURL;
+      readonly attribute USVString blockedURL;
+      readonly attribute USVString destination;
+      readonly attribute boolean   reportOnly;
+    };
+  </pre>
+
+  To <dfn>report violation</dfn> given a <a>Request</a> |request|, a boolean |block|,
+  a boolean |report block|, an <a>integrity policy struct</a> |policy|,
+  and an <a>integrity policy struct</a> |report policy|, do the following:
+
+  1. Assert that |request|'s <a for=request>client</a> is not null.
+  1. Let |settings object| be |request|'s <a for=request>client</a>.
+  1. Let |global| be |settings object|'s <a>global object</a>.
+  1. Assert that |global| is a <a>Window</a> or a <a>WorkerGlobalScope</a>.
+  1. Let |URL| be an empty <a for=/>URL</a>.
```

Comment:
1. All lowercase for the variable. It's only uppercase when it's not the first word.
2. There's no such thing as an empty URL. You probably want null. And maybe an assert after initializing that it's a URL at that point.